### PR TITLE
Transaction View React Components

### PR DESCRIPTION
**Pull request by james-wallis**
### Summary
* Add/Finish React components for the Transaction View:
  * Added a container component to hold the state for both manual and data components (means user won't lose data if they switch accidentally) + display shared features such as submit buttons.
  * Pass args as an array rather than stringifying an array and then converting it back in the extension.
* Remove submit/evaluate transaction commands from context menu and command pallet
* Add Transaction View on left click of Smart Contract command, and context menu for Smart Contract

### Testing
* Added React tests for added components + modified existing tests to handle the move to functional components
* Updated submitTransaction test to handle args being passed as an array in transient data

### Screenshots
**Manual Input**
image

**Transaction Data directory**
image

### CODE REVIEW

**Review by Jakeeyturner — CHANGES_REQUESTED** [review]
Hello,

There seems to be few inconsistencies with the designs in https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/2639#issuecomment-708302390.

1. This seems to have been missed out from these designs, but the 'breadcrumb, but not really a breadcrumb' text at the top, e.g `Transacting with: 2 Org Local Fabric - Org1 / mychannel / mytscontract@0.0.1` should be grey/white so it's not indicating that it's clickable.
2. The peer targeting multi-select should be the second input when 'Manual Input' is selected and the first input when 'Transaction data directory' is selected.
3. The transient data input should be resizable (as it could be a large object).
4. The transaction output doesn't have the image or styling of the designs. The 'Learn more' should probably link off to the `a4` tutorial (this might change depending on what tutorial changes Rob makes).
5. You have a `*` next to submit of the input labels, they're not in the designs. Some of the other input labels aren't the same either.
6. When you associate a transaction directory with a contract the path isn't quite aligned with the 'X' - would be great if we could fix this. Also wondering how that looks with a really large path? Worth looking into. In addition, when I associate a directory, click 'submit' it then asks me if I want to provide a file - this isn't correct. When you associate a directory, we should read all of the transaction data files (there could be more than 1), and put all of the possible transactions (and associated data) in the transaction dropdown.
7. We need to remove submit/evaluate as tree options & not make them callable from the command palette. We also need make the transaction view openable at the smart contract level on right-click. Additionally, a user should be able to left-click on a transaction and the view will open and pre-select the chosen transaction.

Drop me a message if any of these don't make sense / you want to discuss it further!

**Comment by james-wallis** [comment]
Moving the comments above into a list so it's easier for me to track

- [ ] Change breadcrumb to remove colour from slashes
- [x] Peer targeting multi-select. Spoke to Designer Toby and he agreed to have it at the bottom of both so it's consistent for data and manual input + it's optional.
- [ ] Make transient data resizable
- [ ] Add transaction output design when there is no output
- [ ] The 'Learn more' should link to a4 tutorial 
- [ ] Remove the `*` next to input labels
- [ ] Fix incorrect input labels
- [ ] Fix the transaction directory path not aligning with the `X`
- [ ] Submit transaction should read all the data files instead of asking which files to use, and then present all possible transactions (and associated data) in the transaction dropdown
- [ ] Remove submit/evaluate as tree options & not make them callable from the command palette
- [ ] Make the transaction view openable at the smart contract level on right-click
- [ ] A user should be able to left-click on a transaction and the view will open and pre-select the chosen transaction

**Review by Jakeeyturner — CHANGES_REQUESTED** [review]
Screenshot 2020-11-05 at 12 00 24
Can you make 'here' a link which goes to https://github.com/IBM-Blockchain/blockchain-vscode-extension#using-transaction-data-files-to-submit-a-transaction
Screenshot 2020-11-05 at 12 00 27
Can we grey out all this text / make it the default body color to indicate that it's not clickable.
Screenshot 2020-11-05 at 12 02 29
Can we add this icon to the view as well please.

In terms of responsiveness, I think it needs some amendments - we want it to look good when it scales down to roughly half a (Mac) window. Lets not worry about any smaller than that!

File: packages/blockchain-extension/extension/commands/submitTransaction.ts [hunk 1]
```diff
@@ -30,8 +30,58 @@ import { GlobalState, ExtensionData } from '../util/GlobalState';
...
+
+async function createArgs(transactionObject: any): Promise<Array<string>> {
+    let argsString: string;
+    if (transactionObject && transactionObject.args) {
+        if (Array.isArray(transactionObject.args)) {
+            return transactionObject.args;
+        }
+        argsString = transactionObject.args;
+    } else {
+        argsString = await UserInputUtil.showInputBox('optional: What are the arguments to the transaction, (e.g. ["arg1", "arg2"])', '[]');
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/commands/submitTransaction.ts:65]: Surely a user is never going to be asked this as we've disabled submitting/evaluating transactions from the command palette?

**Review by Jakeeyturner — APPROVED** [review]

**Pull request merged**